src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -111,45 +123,43 @@ def to_json_dict(self) -> dict:
             "artifact": self.artifact,
             "version": self.version,
             "packaging": self.packaging,
+            "classifier": self.classifier,
         }
         return ret
 
     @classmethod
-    def from_coord_str(cls, s: str, *, with_2315_workaround=False) -> Coordinate:
-        """Parses from a coordinate string with an optional `packaging` string.
+    def from_coord_str(cls, s: str) -> Coordinate:
+        """Parses from a coordinate string with optional `packaging` and `classifier` coordinates.
 
-        ${organisation}:${artifact}:${version}[:${packaging}]
+        Using Aether's implementation as reference
+        http://www.javased.com/index.php?source_dir=aether-core/aether-api/src/main/java/org/eclipse/aether/artifact/DefaultArtifact.java
 
-        NB: The `with_2315_workaround` flag is for parsing in locations that are affected by
-          https://github.com/coursier/coursier/issues/2315
-        ... which swaps the `version` and `packaging`.
+        ${organisation}:${artifact}[:${packaging}[:${classifier}]]:${version}
         """
-        parts = s.split(":")
-
-        if len(parts) == 4:
-            if with_2315_workaround:
-                version = parts[3]
-                packaging = parts[2]
-            else:
-                version = parts[2]
-                packaging = parts[3]
-        else:
-            version = parts[2]
-            packaging = "jar"
 
-        return cls(
-            group=parts[0],
-            artifact=parts[1],
-            version=version,
-            packaging=packaging,
-        )
+        if parts := Coordinate.REGEX.match(s):
```

Comment:
```suggestion
        parts = Coordinate.REGEX.match(s)
        if parts:
```
Pants currently runs on Python versions down to 3.6, so unfortunately we cannot use the walrus operator yet: this will likely fail in CI.

Reply:
yeah, noticed that, will need to change that for a classic `None` check